bookwyrm_backend/src/main/java/com/bookwyrm/backend/book/controller/BookController.java

```diff
@@ -66,13 +75,115 @@ public ResponseEntity<BookUploadPayload> createBook(
         return ResponseEntity.status(status).body(response);
     }
 
+    //This utility function removes all articles from a list of Strings, it is not case sensitive
+    private ArrayList<String> removeArticles(List<String> unfiltered){
+        ArrayList<String> filtered = new ArrayList<String>(unfiltered);
+        Iterator<String> iter = unfiltered.iterator();
+        while(iter.hasNext()){
+            boolean isArt = false;
+            String word = iter.next();
+            int idx = 0;
+            while(idx < this.ARTICLES.length & !isArt){
+                isArt = this.ARTICLES[idx].equals(word.toLowerCase());
+                idx++;
+            }
+            if(isArt){
+                filtered.remove(word);
+            }
+        }
+        return filtered;
+    }
+
+    //This utility function removes all articles from a Set of Strings, it is not case sensitive
+    //Unfortunately It was necessary to overload this function
+    private HashSet<String> removeArticles(HashSet<String> unfiltered){
+        HashSet<String> filtered = new HashSet<String>(unfiltered);
+        Iterator<String> iter = unfiltered.iterator();
+        while(iter.hasNext()){
+            boolean isArt = false;
+            String word = iter.next();
+            int idx = 0;
+            while(idx < this.ARTICLES.length & !isArt){
+                isArt = this.ARTICLES[idx].equals(word.toLowerCase());
+                idx++;
+            }
+            if(isArt){
+                filtered.remove(word);
+            }
+        }
+        return filtered;
+    }
+
     @CrossOrigin
     @GetMapping("/{title}")
     public ResponseEntity<BookSearchPayload> searchBookByTitle(@PathVariable("title") String title) {
 
         BookSearchPayload response = new BookSearchPayload();
         HttpStatus status = HttpStatus.OK;
-        response.setBookDaoList(bookService.findAllBooksWithTitle(title));
+
+        HashSet<String> uTokens = new HashSet<String>();
```

Comment:
same deal as above `new HashSet<String>` can be `new HashSet<>`

Reply:
uTokens should probably have a better name, because I had to examine the code to figure out the name (which is backwards)